embeddings/config/flair_config.py

```diff
@@ -0,0 +1,175 @@
+    TASK_TRAIN_KEYS: ClassVar[Set[str]] = {
+        "learning_rate",
+        "mini_batch_size",
+        "max_epochs",
+    }
+    document_embedding_cls: str = "FlairDocumentPoolEmbedding"
+    learning_rate: float = 1e-3
+    mini_batch_size: int = 32
+    max_epochs: int = 20
+
+    task_model_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+    task_train_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+    load_dataset_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+
+    def __post_init__(self) -> None:
+        self.task_train_kwargs = self._parse_fields(self.TASK_TRAIN_KEYS)
+
+    @classmethod
+    def from_yaml(cls, path: T_path) -> "FlairBasicConfig":
+        config = read_yaml(path)
+        return cls(**config)
+
+    @classmethod
+    def from_dict(cls, config: Dict[str, Any]) -> "FlairBasicConfig":
+        return cls(**config)
+
+
+@dataclass
+class FlairSequenceLabelingBasicConfig(FlairBasicConfig, FlairSequenceLabelingConfigKeys):
+    hidden_size: int = 256
+    use_crf: bool = True
+    use_rnn: bool = True
+    rnn_type: str = "LSTM"
+    rnn_layers: int = 1
+    dropout: float = 0.0
+    word_dropout: float = 0.05
+    locked_dropout: float = 0.5
+    reproject_embeddings: bool = True
+
+    def __post_init__(self) -> None:
+        self.task_train_kwargs = self._parse_fields(self.TASK_TRAIN_KEYS)
+        self.task_model_kwargs = self._parse_fields(self.TASK_MODEL_KEYS)
+
+    @classmethod
+    def from_yaml(cls, path: T_path) -> "FlairSequenceLabelingBasicConfig":
+        pass
+
+    @classmethod
+    def from_dict(cls, d: Dict[str, Any]) -> "FlairSequenceLabelingBasicConfig":
+        pass
+
+
+@dataclass
+class FlairTextClassificationBasicConfig(FlairBasicConfig, FlairTextClassificationConfigMapping):
+    pooling: str = "mean"
+    fine_tune_mode: str = "none"
+    fine_tune: bool = False
+    cnn_pool_kernels: Tuple[Tuple[int, int], ...] = ((100, 3), (100, 4), (100, 5))
+    hidden_size: int = 256
+    rnn_type: str = "LSTM"
+    rnn_layers: int = 1
+    bidirectional: bool = True
+    dropout: float = 0.0
+    word_dropout: float = 0.05
+    reproject_words: bool = True
+
+    load_model_kwargs: Dict[str, Any] = field(init=False, compare=False, default_factory=dict)
+
+    def __post_init__(self) -> None:
+        task_model_keys = self.TASK_MODEL_KEYS_MAPPING[self.document_embedding_cls]
+        self.task_train_kwargs = self._parse_fields(self.TASK_TRAIN_KEYS)
+        self.task_model_kwargs = self._parse_fields(task_model_keys)
+
+    @classmethod
+    def from_yaml(cls, path: T_path) -> "FlairTextClassificationBasicConfig":
+        pass
+
+    @classmethod
+    def from_dict(cls, d: Dict[str, Any]) -> "FlairTextClassificationBasicConfig":
+        pass
+
+
+@dataclass
+class FlairAdvancedConfig(AdvancedConfig, ABC):
+    DEFAULT_TASK_TRAIN_KWARGS: ClassVar[MappingProxyType[str, Any]] = MappingProxyType(
```

Comment:
I think that we shouldn't provide defaults in `Advanced` configs. For that purpose we have `Basic` configs

Reply:
By providing the default values we, first of all, align these two configs so they store the same values at the beginning and secondly we allow for manipulating the default values of the pipelines which is important e.g. when testing the hps pipeline (e.g. downsampling dataset in datamodule)

Reply:
It this comment still actual? Code has been changed but I still see some artefacts. 

Answer to your reply:  The arguments that you have provided are not convincing to me.
1. I don't think that's needed, Advanced configs should be used beyond basic ones, therefore I don't see any particular reason that at the beginning should provide same results. Maybe we can provide some utility function to do so, but I think it shouldn't be default behaviour.
2. I don't see this argument either. We can define this values in fixtures.

Reply:
If you look at the `test_hps_classification` file then we have:

```
TESTING_DATAMODULE_KWARGS: Dict[str, Any] = deepcopy(
    dict(LightningConfigDefaultKwargs.DEFAULT_DATAMODULE_KWARGS)
)
TESTING_DATAMODULE_KWARGS.update(
    {
        "downsample_train": 0.01,
        "downsample_val": 0.01,
        "downsample_test": 0.05,
    }
)
```

This means we can either keep the defaults for the Lightning pipelines or implement downsample transformation and allow for specifying own preprocessing pipeline in HPS

Reply:
So it is actual but only for Lightning configs

Reply:
But it was temporal solution. I'm not fan of keeping default kwargs because of it.  
 
Solution #1. 
We can allow own preprocessing pipeline in HPS via storing it to file. Therefore we could perform it earlier

Solution #2 
We could add Downsampling transformation to HuggingFacePreprocessingPipeline and add downsampling args to Pipeline